docker-compose.yml

```diff
@@ -1,103 +1,110 @@
 version: "3"
 
 services:
-  setup:
-    container_name: DarkflameSetup
-    build:
-      context: .
-      dockerfile: ./docker/setup.Dockerfile
-    environment:
-      - DATABASE=${MARIADB_DATABASE:-darkflame}
-      - DATABASE_HOST=database
-      - DATABASE_USER=${MARIADB_USER:-darkflame}
-      - DATABASE_PASSWORD=${MARIADB_PASSWORD:-darkflame}
-      - EXTERNAL_IP=${EXTERNAL_IP:-darkflame}
-    volumes:
-      - ${CLIENT_PATH:?missing_client_path}:/client
-      - shared_configs:/docker/
+    setup:
+        container_name: DarkflameSetup
+        build:
+            context: .
+            dockerfile: ./docker/setup.Dockerfile
+        environment:
+            - DATABASE=${MARIADB_DATABASE:-darkflame}
+            - DATABASE_HOST=database
+            - DATABASE_USER=${MARIADB_USER:-darkflame}
+            - DATABASE_PASSWORD=${MARIADB_PASSWORD:-darkflame}
+            - EXTERNAL_IP=${EXTERNAL_IP:-darkflame}
+            - CLIENT_TYPE=${CLIENT_TYPE:-auto}
+            - CLIENT_ROOT_DIR=${CLIENT_ROOT_DIR}
```

Comment:
Now it only gets overwritten if the variable is empty. This was intended so that you can set the path by yourself if you have an unusual setup.

Reply:
Ah ok. Do you want to set a default value of an empty string then here in the `docker-compose.yml` (for example `${CLIENT_ROOT_DIR-}`) to avoid the warning from Docker, and to avoid confusing users?

The current handling for `CLIENT_TYPE == auto` and a non-empty `CLIENT_ROOT_DIR` might be a bit problematic. The checks (such as `if [[ -f "/client/legouniverse.exe" ]]`) seem to ignore `CLIENT_ROOT_DIR`. So this could fail even though `CLIENT_ROOT_DIR` points to a valid directory. Would it make sense to consider the `CLIENT_ROOT_DIR` in the checks, or alternatively to simply fail if the user sets  `CLIENT_TYPE == auto` and a non-empty `CLIENT_ROOT_DIR`?

Reply:
Hm, I expect `CLIENT_TYPE == auto` only if someone don't specifies `CLIENT_ROOT_DIR`.

> Would it make sense to consider the CLIENT_ROOT_DIR in the checks, or alternatively to simply fail if the user sets CLIENT_TYPE == auto and a non-empty CLIENT_ROOT_DIR

I think, this is the way to go :)

Reply:
Ok, I added a warning. I don't think it should fail hard at this point, because it gets overwritten anyway.